### PR TITLE
Implement 1inch data utility

### PR DESCRIPTION
**Pull request by JaredBorders**
Implement a utility function that takes a swap response from the 1inch API and synthswap address as input, processes it, and returns calldata and the function selector designated by 1inch to be called.

## Description
* Create `getFormatedSwapData()`
* Publish module to npm: https://www.npmjs.com/package/@jaredborders/synthswap-utils
* Test `getFormatedSwapData()`

## Related issue(s)
Closes [Build Utility for Fetching 1inch Swap Data](https://github.com/Kwenta/synthswap/issues/16)

## Motivation and Context
Reduce cognitive load on front-end relating to 1inch swap mechanics/data processing

### CODE REVIEW

File: SynthSwap-Util/package/index.js [hunk 1]
```diff
@@ -0,0 +1,1142 @@
...
+// response parameters:
+// https://docs.1inch.io/docs/aggregation-protocol/api/swap-params#description-of-response-parameters
+
+// Swagger API: 
+// https://docs.1inch.io/docs/aggregation-protocol/api/swagger
+
+/// getFormatedSwapData returns data required by SynthSwap to execute swaps
+/// @param response: from /v4.0/10/swap
+/// @param synthwapAddress: address of deployed SynthSwap contract
+/// @returns Object: {functionSelector, data}
```
Comment (JChiaramonte7) [inline SynthSwap-Util/package/index.js:847]: Is this the 1inch `AggregationRouterV4.sol` function selector or `Synthswap` function selector?

File: SynthSwap-Util/package/index.js [hunk 2]
```diff
@@ -0,0 +1,1142 @@
+const ethers = require('ethers');
+const abiDecoder = require('abi-decoder');
+
+// ABI of 1inch AggregationRouterV4 
+// @ address 0x1111111254fb6c44bAC0beD2854e76F90643097d on Ethereum Mainnet
+const AggregationRouterV4ABI = [
```
Comment (JChiaramonte7) [inline SynthSwap-Util/package/index.js:6]: Probably better to put in its own file for readability of this file.

File: SynthSwap-Util/package/package.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  "keywords": [
+    "synthswap"
+  ],
+  "author": "JaredBorders",
+  "license": "ISC",
+  "dependencies": {
+    "abi-decoder": "^2.4.0",
+    "ethers": "^5.6.5"
+  }
+}
```
Comment (JChiaramonte7) [inline SynthSwap-Util/package/package.json:18]: Really was thinking we don't make this a subset of synthswap, but we bundle all of synthswap as an NPM package. Imagine you wanted to integrate synthswap with another project or needed access to the interface. Would be nice just to `npm install @kwenta/synthswap`

**Review by JChiaramonte7 — CHANGES_REQUESTED** [review]
Thinking we make all of synthswap an npm package then we don't need to have a separate `SynthSwap-Util/` folder.

File: SynthSwap-Util/package/index.js (hunk 1 above)
Comment (JaredBorders) [reply SynthSwap-Util/package/index.js:847]: AggregationRouterV4.sol

**Comment by JChiaramonte7** [comment]
I still see `SynthSwap-Util/` is that still used?

**Comment by JaredBorders** [comment]
@JChiaramonte7 I just removed it. I was using it to create utils/ and forgot to delete it after

**Review by JChiaramonte7 — APPROVED** [review]

**Pull request merged**